### PR TITLE
Student record

### PR DESCRIPTION
**Pull request by rodriguez-maria**
This is part of the 5 problems for assignment 2.

### CODE REVIEW

File: src/dev/StudentAttendanceRecordI.java [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+ * Author: Maria Rodriguez
+ * Date: October 2, 2017
+ */
+
+package dev;
+
+public class StudentAttendanceRecordI {
+	public static boolean isRewarded(String str)
+	{
+		if(str == null || str.length()==0)
```
Comment (rfaisal) [inline src/dev/StudentAttendanceRecordI.java]: Let's do the following to check null or empty from now on:

if(str == null || str.isEmpty()){....}

File: src/dev/StudentAttendanceRecordI.java [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+		{
+			return false;
+		}
+		
+		int absent = 0;
+		int late = 0;
```
Comment (rfaisal) [inline src/dev/StudentAttendanceRecordI.java]: The name late is confusing it should be continuousLate or something like that

File: src/dev/StudentAttendanceRecordI.java [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+				break;
+			}
+			case 'L':
+			{
+				if(i > 0 && str.charAt(i-1) == 'L')
+				{
+					late += 1;
+				}
+				else
+				{
```
Comment (rfaisal) [inline src/dev/StudentAttendanceRecordI.java]: why do you need this condition

File: src/dev/StudentAttendanceRecordI.java [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+		int absent = 0;
+		int late = 0;
+		
+		for(int i=0; i<str.length(); i++)
+		{
+			switch(str.charAt(i))
+			{
+			case 'A': 
+			{
+				absent += 1;
```
Comment (rfaisal) [inline src/dev/StudentAttendanceRecordI.java]: always absent++

File: src/test/StudentAttendanceRecordITests.java [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+public class StudentAttendanceRecordITests {
+
+	@Test
+	public void nullEmptyCases() {
+		assertFalse(StudentAttendanceRecordI.isRewarded(null));
+		assertFalse(StudentAttendanceRecordI.isRewarded(""));
+	}
+	
+	@Test
+	public void normalCases() {
```
Comment (rfaisal) [inline src/test/StudentAttendanceRecordITests.java]: should LLLLPPPPPP rewarded or not?

File: src/test/StudentAttendanceRecordITests.java [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+		assertTrue(StudentAttendanceRecordI.isRewarded("PPPPPPP"));
+		assertTrue(StudentAttendanceRecordI.isRewarded("APP"));
+		assertTrue(StudentAttendanceRecordI.isRewarded("A"));
+		assertTrue(StudentAttendanceRecordI.isRewarded("P"));
+		assertTrue(StudentAttendanceRecordI.isRewarded("L"));
+		assertFalse(StudentAttendanceRecordI.isRewarded("PPALLL"));
+		assertFalse(StudentAttendanceRecordI.isRewarded("LLLLLLLLL"));
+		assertFalse(StudentAttendanceRecordI.isRewarded("PAPPPPPPAPPP"));
+		assertFalse(StudentAttendanceRecordI.isRewarded("LPPPPPPPLLLAPPPP"));
+	}
```
Comment (rfaisal) [inline src/test/StudentAttendanceRecordITests.java]: I think your program would return false for LPPLLPPLL which is not corrent

File: src/dev/StudentAttendanceRecordI.java (hunk 1 above)
Comment (rodriguez-maria) [reply src/dev/StudentAttendanceRecordI.java]: Done

File: src/dev/StudentAttendanceRecordI.java (hunk 2 above)
Comment (rodriguez-maria) [reply src/dev/StudentAttendanceRecordI.java]: Done!

File: src/dev/StudentAttendanceRecordI.java (hunk 4 above)
Comment (rodriguez-maria) [reply src/dev/StudentAttendanceRecordI.java]: Done!

File: src/test/StudentAttendanceRecordITests.java (hunk 5 above)
Comment (rodriguez-maria) [reply src/test/StudentAttendanceRecordITests.java]: Should not

File: src/test/StudentAttendanceRecordITests.java (hunk 6 above)
Comment (rfaisal) [reply src/test/StudentAttendanceRecordITests.java]: ignore

**Pull request merged**